src/python/py/models/builder.py

```diff
@@ -713,6 +722,60 @@ def make_matmul_int4(self, matmul, basename, root_input, **kwargs):
 
         return name
 
+    def make_dequantize_linear(self, dequantize_name, quantized_op):
+        # Input weights are quantized, save quantized MatMul numpy weights for onnx model
+        qweight = dequantize_name[1:].replace("/", ".") + ".qweight"
+        qweight_npy = quantized_op.qweight.detach().numpy()
+        qweight_npy = qweight_npy.reshape(*qweight_npy.shape[:-2], qweight_npy.shape[-2] * qweight_npy.shape[-1])
+        self.make_external_tensor(qweight_npy, qweight, True)
+
+        scales = dequantize_name[1:].replace("/", ".") + ".scales"
+        scales_npy = quantized_op.scales.detach().numpy().astype(self.to_numpy_dtype[self.io_dtype])
+        scales_npy = scales_npy.reshape(*qweight_npy.shape[:-1], qweight_npy.shape[-1] * 2 // quantized_op.group_size)
+        self.make_external_tensor(scales_npy, scales)
+
+        dequantize_inputs = [qweight, scales]
+
+        if hasattr(quantized_op, "qzeros") and quantized_op.qzeros is not None:
+            zeros = dequantize_name[1:].replace("/", ".") + ".qzeros"
+            zeros_npy = quantized_op.qzeros.detach().numpy()
+            zeros_npy = zeros_npy.reshape(*qweight_npy.shape[:-1], qweight_npy.shape[-1] // quantized_op.group_size)
+            self.make_external_tensor(zeros_npy, zeros, True)
+            dequantize_inputs.append(zeros)
+
+        dequantize_output = f"{dequantize_name}/output_0"
+        self.make_node("DequantizeLinear", inputs=dequantize_inputs, outputs=[dequantize_output], name=dequantize_name, block_size=quantized_op.group_size)
```

Comment:
The default axis is 1. it is 0 in this case.

Reply:
The correct axis is actually `1` here, I can make it explicit by hardcoding the axis to the last dimension, but the second dimension is the one that's blocked, not the first one.